### PR TITLE
fix(connection): handle canceled tasks correctly

### PR DESCRIPTION
**Pull request by mxschmitt**
Was hanging before

Closes #536

### CODE REVIEW

File: tests/async/test_issues.py [hunk 1]
```diff
@@ -30,3 +34,18 @@ async def test_issue_195(playwright, browser):
...
+        create_task(page.wait_for_selector("input")),
+        create_task(page.wait_for_selector("#will-never-resolve")),
+    }, return_when=FIRST_COMPLETED)
+    assert len(done) == 1
+    assert len(pending) == 1
+    for task in pending:
+        task.cancel()
+        with suppress(CancelledError):
+            await task
+    assert list(pending)[0].cancelled() is True
```
Comment (kumaraditya303) [inline tests/async/test_issues.py:51]:
```suggestion
    assert list(pending)[0].cancelled()
```

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]

File: tests/async/test_issues.py [hunk 2]
```diff
@@ -30,3 +34,18 @@ async def test_issue_195(playwright, browser):
...
+    await page.set_content("<input />")
+    done, pending = await wait({
+        create_task(page.wait_for_selector("input")),
+        create_task(page.wait_for_selector("#will-never-resolve")),
+    }, return_when=FIRST_COMPLETED)
+    assert len(done) == 1
+    assert len(pending) == 1
+    for task in pending:
+        task.cancel()
+        with suppress(CancelledError):
```
Comment (kumaraditya303) [inline tests/async/test_issues.py:49]:
```suggestion
        with pytest.raises(CancelledError):
```

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]

File: tests/async/test_issues.py (hunk 2 above)
Comment (kumaraditya303) [reply tests/async/test_issues.py:49]: This ensures that task was cancelled 🚀

Comment (mxschmitt) [reply tests/async/test_issues.py]: good catch, simplified it too.

**Review by kumaraditya303 — APPROVED** [review]

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**